### PR TITLE
GG-23802 Removed duplicate error causes.

### PR DESCRIPTION
**Pull request by vsisko**

### CODE REVIEW

File: modules/web-console/frontend/app/services/ErrorParser.service.ts [hunk 1]
```diff
@@ -72,7 +72,8 @@ export default class {
                     const msgEnd = err.message.indexOf('[', msgStart);
                     const cause = err.message.substring(msgStart, msgEnd >= 0 && msgEnd < causeEndLine ? msgEnd : causeEndLine);
 
-                    causes.unshift(cause);
+                    if (causes && causes[0] !== cause)
```
Comment (akuznetsov-gridgain) [inline modules/web-console/frontend/app/services/ErrorParser.service.ts]: Lets cover ErrorParser.service.ts  with tests.

Comment (Klaster1) [reply modules/web-console/frontend/app/services/ErrorParser.service.ts]: Second this, at least add a test for the fix.

Comment (vsisko) [reply modules/web-console/frontend/app/services/ErrorParser.service.ts]: Test has been added.

File: modules/web-console/frontend/app/services/ErrorParser.spec.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+const parser = new ErrorParser(new JavaTypes());
+
+import { assert } from 'chai';
+
+const EXPECTED_CAUSES = ['Root cause', 'Duplicate cause', 'Final cause'];
+
+const TEST_ERROR = {
+    className: 'test.Exception1',
+    message: 'Failed to handle request: [req=EXE, taskName=test.TaskName, params=[], err=Final cause, trace=...]\n' +
+        '\tat test.Class1.function(Class1.java:1)\n' +
```
Comment (Klaster1) [inline modules/web-console/frontend/app/services/ErrorParser.spec.ts]: Please use backticks for multiline strings.

File: modules/web-console/frontend/app/services/ErrorParser.spec.ts [hunk 3]
```diff
@@ -25,28 +25,28 @@ const EXPECTED_CAUSES = ['Root cause', 'Duplicate cause', 'Final cause'];
...
+'Caused by: class test.Exception3: Duplicate cause
+    at test.Class3.function(Class3.java:1)
+    ... 14 more
+'Caused by: class test.Exception4: Duplicate cause
+    at test.Class4.function(Class4.java:1)
+    ... 29 more
+'Caused by: class test.Exception5: Root cause
+    at test.Class5.function(Class5.java:1)
+    ... 40 more
+']`
```
[the remark this answers is not in the document]
Comment (vsisko) [reply modules/web-console/frontend/app/services/ErrorParser.spec.ts]: Removed wrong symbols.

File: modules/web-console/frontend/app/services/ErrorParser.spec.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    test('Error parsing', () => {
+        assert.equal(FULL_ERROR_MSG, parser.extractFullMessage(TEST_ERROR));
+
+        const parsed = parser.parse(TEST_ERROR, 'Test: ');
+
+        assert.equal(ERROR_MSG, parsed.message);
+
+        assert.deepEqual(EXPECTED_CAUSES, parsed.causes);
+    });
+});
```
[the remark this answers is not in the document]
Comment (vsisko) [reply modules/web-console/frontend/app/services/ErrorParser.spec.ts]: Line added. IDE configured.

**Pull request merged**